cmd/pulumi-test-language/providers/component_provider.go

```diff
@@ -537,20 +541,22 @@ func (p *ComponentProvider) constructComponentCallable(
 ) (plugin.ConstructResponse, error) {
 	// Register the parent component.
 	parent, err := monitor.RegisterResource(ctx, &pulumirpc.RegisterResourceRequest{
-		Type: "component:index:ComponentCallable",
-		Name: req.Name,
+		Type:     "component:index:ComponentCallable",
+		Name:     req.Name,
+		Provider: req.Options.Providers["component"],
 	})
 	if err != nil {
 		return plugin.ConstructResponse{}, fmt.Errorf("register parent component: %w", err)
 	}
 
 	// Register a child resource, parented to the component we just created.
 	child, err := monitor.RegisterResource(ctx, &pulumirpc.RegisterResourceRequest{
-		Type:    "component:index:Custom",
-		Custom:  true,
-		Name:    req.Name + "-child",
-		Parent:  parent.Urn,
-		Version: "13.3.7",
+		Type:     "component:index:Custom",
+		Custom:   true,
+		Name:     req.Name + "-child",
+		Parent:   parent.Urn,
+		Version:  "13.3.7",
+		Provider: req.Options.Providers["component"],
```

Comment:
All these changes modify this test provider so that it respects and uses the `Providers` resource option.

Reply:
Is `req.Options.Providers["component"]` always set, even when the user did not specify it explicitly?

Reply:
No

Reply:
No, but in those cases we'll send `nil`, which is indistinguishable from not setting it (the default, which will trigger a default provider request).

Reply:
And I believe (given other passing conformance tests) that `req.Providers` is guaranteed to be non-`nil`. But can add the checks if we think not